src/libpriv/rpmostree-postprocess.c

```diff
@@ -356,21 +355,15 @@ process_kernel_and_initramfs (int            rootfs_dfd,
     return FALSE;
 
   RpmOstreePostprocessBootLocation boot_location =
-    RPMOSTREE_POSTPROCESS_BOOT_LOCATION_BOTH;
+    RPMOSTREE_POSTPROCESS_BOOT_LOCATION_NEW;
   const char *boot_location_str = NULL;
   if (!_rpmostree_jsonutil_object_get_optional_string_member (treefile,
```

Comment:
Hmm, let's just trust that serde checked this and take advantage of this cleanup to completely drop this section?

Reply:
I now added the next commit to this PR, so we want to keep this section.  Don't think it's really worth changing from this from `GError` to `g_assert()` right now.  Ideally down the line we export the treefile as a struct that the C side can just read but we have some nontrivial work to get there.